homeassistant/components/filter/sensor.py

```diff
@@ -488,7 +488,7 @@ class RangeFilter(Filter, SensorEntity):
     def __init__(
         self,
         entity,
-        precision: int | None = DEFAULT_PRECISION,
+        precision,
```

Comment:
Any reason not to keep `int | None` type definition?  Or is it standard to not include it if you're not including a default value?

Reply:
If I include the type definition mypy complains :(

Reply:
I think mypy probably complains for a reason...
I started to look into it and I found it very awkward that functions only had partial type hints, and some even had invalid type hints.
I suggest that we fix the type hints first: #86141